### PR TITLE
Update deposits REST API docs with decoupled deposit changes

### LINKED ISSUE #7847: Update deposits REST API documentation to show changes when transactions are decoupled from instant & estimated deposits
As we decouple transactions from instant (manual) and estimated deposits on the server, this will impact the API from the `/deposits/*` endpoints.

We need to understand and communicate these changes to consumers/clients of these endpoints.

This issue aims to document these changes via the deposits REST API documentation introduced in #7840.

See project thread: paJDYF-aY6-p2
See discussion on removing estimated deposits: paJDYF-bD7-p2

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #7847

#### Changes proposed in this Pull Request

This PR updates the deposits REST API documentation to reflect the "decoupled deposits" changes deployed on Jan 23 2024 in server PR 4500.

> [!note]
> I've uploaded a preview of this PR's version of the docs [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits).

- [x] `GET deposits/overview-all`
  * Response `deposit.next_scheduled` is no longer returned.
  * Response `balance.pending[].deposits_count` is no longer returned.
  * Response `balance.instant[].transaction_ids` is no longer returned.
- [x] `GET deposits/overview`
  * Response `next_deposit` is no longer returned.
  * Response `balance.pending.deposits_count` is no longer returned.
  * Response `instant_balance.transaction_ids` is no longer returned.
- [x] `GET deposits` 
  * `estimated` deposits are no longer returned.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.
- [x] `GET deposits/summary`
  * `estimated` deposits are no longer returned.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.
- [x] `GET deposits/(?P<deposit_id>\w+)`
  * `estimated` deposits are no longer returned, will return status `404` in this case.
- [x] `POST deposits`
  * Request body `transaction_ids` property is deprecated.
  * Request body now accepts optional but recommended `currency` property.
  * Implemented in PR #7828
- [x] `POST deposits/download`
  * `estimated` deposits are no longer included in CSV export.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.

See project thread: paJDYF-aY6-p2
See discussion on removing estimated deposits: paJDYF-bD7-p2

#### Testing instructions

* Build the docs locally (requires Docker)
  * `cd docs/rest-api`
  * `./build.sh` to build the docs via a Docker container.
  * `npx serve build` to serve the built HTML at `localhost:3000`
* Alternatively, I've uploaded this PR's version of the docs manually [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits).
* Find and navigate to the **Deposits** section via the sidebar
* Ensure the changes and deprecations to the endpoint responses are correct as per the plan in server epic https://github.com/Automattic/woocommerce-payments-server/issues/4296

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times.
- [x] Covered with tests  __N/A__
- [x] Tested on mobile __N/A__

**Post merge**

- [x] Publish the docs via the [deploy-api-docs.yml GH action](https://github.com/Automattic/woocommerce-payments/actions/workflows/deploy-api-docs.yml) to [automattic.github.io/woocommerce-payments](https://automattic.github.io/woocommerce-payments/) once merged to develop.
- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : **QA Testing Not Applicable**
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. **N/A**
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **N/A**

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7848` or branch name `fix/7847-update-deposits-api-docs-estimated-deposits-rm` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7847-update-deposits-api-docs-estimated-deposits-rm) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9e68172c5bb677c6dc4b6d0ecdca2db75eb00f3a
- Build time: 2024-01-25 04:02:48 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by shendy-a8c — COMMENTED** [review]
The updated docs does show the expected changes correctly as far as I can tell. 

However, I wonder if we should just remove the removed properties instead of showing strike-throughs and "Deprecated since 7.0.0" part because it'd be hard to maintain.

What do you think?

**Comment by Jinksi** [comment]
> However, I wonder if we should just remove the removed properties instead of showing strike-throughs and "Deprecated since 7.0.0" part because it'd be hard to maintain.

I thought it would be easier to illustrate the changes for communicative purposes within the context of this project. What do you think about leaving the strikethrough and deprecation notes for now, before removing them when the time comes to merge/release?

**Comment by shendy-a8c** [comment]
Sure. I agree that they give clear visible changes. Also, should it be 7.0.0 or 7.1.0?

If we are pushing server changes on Jan 24th or later, we should say 7.1.0 because it's scheduled to be released on Jan 24th.

**Review by shendy-a8c — DISMISSED** [review]
Sure. I agree that they give clear visible changes. Also, should it be 7.0.0 or 7.1.0?

If we are pushing server changes on Jan 24th or later, we should say 7.1.0 because it's scheduled to be released on Jan 24th.

**Comment by Jinksi** [comment]
> Also, should it be 7.0.0 or 7.1.0?

Thanks @shendy-a8c – updated 7.0.0 → 7.1.0 in dc9227b. Changes visible on docs preview [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits).

**Comment by Jinksi** [comment]
To update this PR's progress: I'll be:

1. validating these changes once the server deployment of https://github.com/Automattic/woocommerce-payments-server/pull/4500 has occurred.
2. removing the redundant/deprecated fields from the docs, as discussed here: https://github.com/Automattic/woocommerce-payments/pull/7848#pullrequestreview-1802840615.
3. re-requesting a review

**Comment by Jinksi** [comment]
I've validated the docs against the production server deployment of decouple deposits API changes.

I've removed the deprecated fields and values as per the discussion above.

See preview [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits). Ready for re-review.

File: docs/rest-api/source/includes/wp-api-v3/deposits.md [hunk 1]
```diff
@@ -481,7 +413,7 @@ Fetches a deposit by ID.
 
 If a deposit is found for the provided ID, the response will return a [**Deposit**](#deposit-object) object.
 
-If no deposit is found for the provided ID, the response will be an empty array.
+If no deposit is found for the provided ID, the response will return a `500` status code.
```
Comment (brucealdridge) [inline docs/rest-api/source/includes/wp-api-v3/deposits.md:416]: NB: The HTTP response code was changed to `404` on the server, the client still responds with `500`.

**Review by brucealdridge — APPROVED** [review]
Reviewed test site and removals. 👍

File: docs/rest-api/source/includes/wp-api-v3/deposits.md (hunk 1 above)
Comment (haszari) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md:416]: Interesting … I like the idea that we work towards no 500 errors from any of our APIs.

Comment (Jinksi) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md:416]: Yeh this came up in #4342 which implemented the server returning `404`, only for client to swap it to a `500`.

For now, the docs should reflect the current state of the REST API but we can open a new issue to ensure `404`'s are passed through to client REST API responses.

Comment (haszari) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md:416]: Nice, yea I'm thinking long term here. Also might be a cool way into more monitoring of our APIs – e.g. monitoring can tell us when we return 500, we can work toward no 500s (maybe in API paths that we own).

**Comment by Jinksi** [comment]
Docs deployed to https://automattic.github.io/woocommerce-payments/#deposits, workflow https://github.com/Automattic/woocommerce-payments/actions/runs/7661809270 was automatically triggered.

**Pull request merged**